### PR TITLE
Adding integration test cases

### PR DESCRIPTION
**Pull request by idavis**
@swernli Here are the other test files you had. I'm not sure what test cases are handled other than Shor in the bitcode. We can merge these as needed. I wanted to save them off as a todo.

### CODE REVIEW

File: runner/tests/tests.rs [hunk 1]
```diff
@@ -13,6 +13,30 @@ fn test_choi_jamiolkowski_isomorphism() -> Result<(), String> {
     run_bitcode(bitcode, None)
 }
 
+#[test]
+fn test_full_qir_simple() -> Result<(), String> {
+    let bitcode = include_bytes!("resources/full-qir.bc");
+    run_bitcode(bitcode, Some("QIR_App_Test__Simple"))
```
Comment (billti) [inline runner/tests/tests.rs:19]: What do the strings (e.g. 'QIR_App_Test__Simple`) here represent?

File: runner/tests/tests.rs [hunk 2]
```diff
@@ -13,6 +13,30 @@ fn test_choi_jamiolkowski_isomorphism() -> Result<(), String> {
...
+
+#[test]
+fn test_full_qir_other() -> Result<(), String> {
+    let bitcode = include_bytes!("resources/full-qir.bc");
+    run_bitcode(bitcode, Some("QIR_App_Test__Other"))
+}
+
+#[test]
+fn test_qir_testing() -> Result<(), String> {
+    let bitcode = include_bytes!("resources/qir-testing.bc");
```
Comment (billti) [inline runner/tests/tests.rs:30]: Do we have any description of that this file (e.g. `qir-testing.bc`) contains or what this is checking for specifically? (i.e. does this file contain all valid opcodes or something?)

**Review by billti — APPROVED** [review]
Few minor comments/questions. Else thanks!

File: runner/tests/tests.rs (hunk 1 above)
Comment (swernli) [reply runner/tests/tests.rs:22]: These strings are the names of the entry point function to execute. It's an optional parameter that is only required if there is more than one entry point function in the QIR program.

File: runner/tests/tests.rs (hunk 2 above)
Comment (swernli) [reply runner/tests/tests.rs]: I've refactored some of the tests and added descriptions for each to help clarify!

**Pull request merged**